### PR TITLE
Add support for k8s 1.23 and ArgoCD 2.4 client libraries in Arlon

### PR DESCRIPTION
**Pull request by cruizen**
Fixes https://github.com/arlonproj/arlon/issues/164
Aha! Link: https://pf9.aha.io/features/ARLON-274

For validation, just checked that the code compiles. Should run some tests ideally to validate. 

We can also upgrade to k8s 1.24 libraries if we wish (haven't already checked if the code compiles fine with that).

### CODE REVIEW

**Comment by cruizen** [comment]
After upgrading to ArgoCD 2.4, `go mod tidy` was failing with the following error 
```
Triloks-iMac:arlon trilok$ go mod tidy
go mod tidy: error loading go 1.16 module graph: github.com/argoproj/argo-cd/v2@v2.4.11 requires
	gomodules.xyz/notify@v0.1.0 requires
	github.com/appscode/go@v0.0.0-20190808133642-1d4ef1f1c1e0 requires
	github.com/flosch/pongo2@v0.0.0-20181225140029-79872a7b2769 requires
	github.com/go-check/check@v1.0.0-20180628173108-788fd7840127: invalid pseudo-version: major version without preceding tag must be v0, not v1

If reproducibility with go 1.16 is not needed:
	go mod tidy -compat=1.17
For other options, see:
	https://golang.org/doc/modules/pruning
```
And needs to be run as `go mod tidy -compat=1.17`
i.e. the flag `-compat=1.17` is required to `go mod tidy` as things stand.

File: go.mod [hunk 1]
```diff
@@ -133,71 +134,74 @@ require (
...
 	k8s.io/utils v0.0.0-20211116205334-6203023598ed // indirect
-	sigs.k8s.io/kustomize/api v0.8.11 // indirect
+	sigs.k8s.io/json v0.0.0-20211020170558-c049b76a60c6 // indirect
+	sigs.k8s.io/kustomize/api v0.10.1 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.1 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
 replace (
 	github.com/dgrijalva/jwt-go => github.com/golang-jwt/jwt/v4 v4.4.2
```
Comment (Rohitrajak1807) [inline go.mod:180]:
```suggestion
	
```
Compiles fine without the replace, maybe we can incorporate this change too?

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Build the manager binary
-FROM golang:1.17.13 as builder
+FROM golang:1.19 as builder
```
Comment (Rohitrajak1807) [inline Dockerfile:2]: We're using 1.18 in the mod file, so will this be alright?

Comment (Rohitrajak1807) [reply Dockerfile:2]: Marking as resolved after our conversation.

**Review by Rohitrajak1807 — APPROVED** [review]

**Review by bcle — APPROVED** [review]
If you can confirm that the crash described here: https://platform9.slack.com/archives/C02SEAQ0NB1/p1655428356901019 is no longer an issue when upgrading to golang 1.18+, then this is approved.

File: go.mod (hunk 1 above)
Comment (cruizen) [reply go.mod]: Actually, this causes the following change in go.sum that will re-introduce the CVE. 
```
+github.com/dgrijalva/jwt-go v3.2.0+incompatible/go.mod h1:E3ru+11k8xSBh+hMPgOLZmtrrCbhqsmaPHjLKYnJCaQ=
.......
-github.com/golang-jwt/jwt/v4 v4.4.2/go.mod h1:m21LjoU+eqJr34lmDMbreY2eSTRJ1cv77w39/MY0Ch0=
```

Comment (Rohitrajak1807) [reply go.mod:184]: Alright let's keep the replace.

**Comment by cruizen** [comment]
> If you can confirm that the crash described here: https://platform9.slack.com/archives/C02SEAQ0NB1/p1655428356901019 is no longer an issue when upgrading to golang 1.18+, then this is approved.

I can confirm that we are not seeing this problem in a controller built with golang v1.18 or v1.19

**Pull request merged**